### DISCUSSION TITLE
System.InvalidOperationException: 'LogFactory must be Initialized before creating any logger' when I try to get a ILogger instance.

### DISCUSSION THREAD

**Discussion by ataparia** [roubachof/MetroLog#30] [General]
I have a specific requirement where I need to hide the implementation details of MetroLogs behind an interface. 
I create a LoggingService where I expose different methods like Info, Error, Warn etc and internally all these methods calls the MetroLogs.Maui apis.

However, I get the following error message when I try to get an instance of ILogger using the following code:
LoggerFactory.GetLogger(<categoryName>);

I did initialize the MauiProgram.cs with the code defined on the github page.

Am I missing something?

**Comment by roubachof** [comment]
Same answer that here: https://github.com/roubachof/MetroLog/discussions/29

**Comment by roubachof** [comment]
closing for duplicate
